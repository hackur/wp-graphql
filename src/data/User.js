import { toGlobalId } from 'graphql-relay';

const path = process.env.WP_USERS_ENDPOINT || 'wp/v2/users';

class User {
  getID() {
    return toGlobalId(this.constructor.name, this.id);
  }

  static getEndpoint() {
    return path;
  }
}

export default User;
