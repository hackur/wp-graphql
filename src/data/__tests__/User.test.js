import User from 'data/User';

describe('Test User data access', () => {
  test('Get endpoint', () => {
    expect(User.getEndpoint()).toMatchSnapshot();
  });

  test('Load a user', async () => {
    const user = await User.load(1);
    expect(user.getID()).toMatchSnapshot();
    expect(user).toMatchSnapshot();
  });

  test('Load a user by slug', async () => {
    const user = await User.loadBySlug('admin');
    expect(user).toMatchSnapshot();
  });
});
