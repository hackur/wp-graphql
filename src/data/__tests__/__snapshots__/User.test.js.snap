// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test User data access Get endpoint 1`] = `"wp/v2/users"`;

exports[`Test User data access Load a user 1`] = `"VXNlcjox"`;

exports[`Test User data access Load a user 2`] = `
User {
  "id": 1,
}
`;

exports[`Test User data access Load a user by slug 1`] = `
User {
  "slug": "admin",
}
`;
